sdk/cosmosdb/cosmos/src/client/ChangeFeed/ChangeFeedForEpkRange.ts

```diff
@@ -0,0 +1,383 @@
+    } else {
+      for (const cToken of contToken.Continuation) {
+        const queryRange = new QueryRange(cToken.minInclusive, cToken.maxExclusive, true, false);
+        try {
+          const overLappingRanges = await this.partitionKeyRangeCache.getOverlappingRanges(
+            this.url,
+            queryRange,
+            this.diagnosticContext
+          );
+          for (const overLappingRange of overLappingRanges) {
+            // check if the epk range present in continuation token entirely covers the overlapping range.
+            // If yes, minInclusive and maxExclusive of the overlapping range will be set.
+            // If no, i.e. there is only partial overlap, epkMinHeader and epkMaxHeader are set as min and max of overlap.
+            // This will be used when we make a call to fetch change feed.
+            const [epkMinHeader, epkMaxHeader] = await checkEpkHeaders(cToken, overLappingRange);
+            const feedRange: ChangeFeedRange = new ChangeFeedRange(
+              overLappingRange.minInclusive,
+              overLappingRange.maxExclusive,
+              cToken.continuationToken,
+              epkMinHeader,
+              epkMaxHeader
+            );
+            this.queue.enqueue(feedRange);
+          }
+        } catch (err) {
+          throw new ErrorResponse(err.message);
+        }
+      }
+
+      return true;
+    }
+  }
+
+  /**
+   * Change feed is an infinite feed. hasMoreResults is always true.
+   */
+  get hasMoreResults(): boolean {
+    return true;
+  }
+
+  /**
+   * Gets an async iterator which will yield pages of results from Azure Cosmos DB.
+   *
+   * Keeps iterating over the feedranges and checks if any feed range has new result. Keeps note of the last feed range which returned non 304 result.
+   *
+   * When same feed range is reached and no new changes are found, a 304 (not Modified) is returned to the end user. Then starts process all over again.
+   */
+  public async ReadNextAsync(): Promise<ChangeFeedIteratorResponse<Array<T & Resource>>> {
+    // stores the last feedRange for which statusCode is not 304 i.e. there were new changes in that feed range.
+    let firstNotModifiedFeedRange: [string, string] = undefined;
+    let result: ChangeFeedIteratorResponse<Array<T & Resource>>;
+    do {
+      const [processedFeedRange, response] = await this.fetchNext();
+      result = response;
+      if (result !== undefined) {
+        {
+          if (firstNotModifiedFeedRange === undefined) {
+            firstNotModifiedFeedRange = processedFeedRange;
+          }
+          // move current feed range to end of queue to fetch result of next feed range.
+          // This is done to fetch changes in breadth first manner and avoid starvation.
+          this.queue.moveFirstElementToTheEnd();
+          // check if there are new results for the given feed range.
+          if (result.count > 0) {
+            firstNotModifiedFeedRange = processedFeedRange;
```

Comment:
Shouldn't this be setting firstNotModifiedFeedRange to undefined?

Reply:
Removed this line, it was redundant as it automatically gets set to undefined when new ReadNextAsync() call is made.